### ISSUE TITLE
Visual Studio 2017 warning

### ISSUE THREAD

**Issue by petko** [nlohmann/json#1784]
With release 3.7.0, in VS2017 (15.9.16), I get the following error when using the library in my tests (it is reported as error, but the projects builds OK):
```
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E2512	the argument to a feature-test macro must be a simple identifier	Tests	[path]\json.hpp	1274	
```

I hope it's not a duplicate, but haven't found it reported..

**Comment by t-b** [comment]
I'm seeing that as well.

The line in question is

https://github.com/nlohmann/json/blob/d187488e0db0533bdd7c53ec0c687ca1745b8b9e/single_include/nlohmann/json.hpp#L1270-L1280

and that is a verbatim copy from 
https://github.com/nemequ/hedley/blob/c395642679d4b79cd54d4f4a3ecb21fdf544513e/hedley.h#L1171-L1181.

So I think that bug should be actually reported to https://github.com/nemequ/hedley.

**Comment by nlohmann** [comment]
I agree. This issue is not related to this library, but to Hedley.

**Comment by nlohmann** [comment]
Closed with #1799.
